samples/AzureSample.ResourceManager.Sample/src/Generated/AvailabilitySetResource.Serialization.cs

```diff
@@ -13,14 +13,17 @@ namespace AzureSample.ResourceManager.Sample
 {
     public partial class AvailabilitySetResource : IJsonModel<AvailabilitySetData>
     {
+        private static AvailabilitySetData s_dataDeserializationInstance;
+        private static AvailabilitySetData DataDeserializationInstance => s_dataDeserializationInstance ??= new();
+
         void IJsonModel<AvailabilitySetData>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options) => ((IJsonModel<AvailabilitySetData>)Data).Write(writer, options);
 
-        AvailabilitySetData IJsonModel<AvailabilitySetData>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options) => ((IJsonModel<AvailabilitySetData>)Data).Create(ref reader, options);
+        AvailabilitySetData IJsonModel<AvailabilitySetData>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options) => ((IJsonModel<AvailabilitySetData>)DataDeserializationInstance).Create(ref reader, options);
 
         BinaryData IPersistableModel<AvailabilitySetData>.Write(ModelReaderWriterOptions options) => ModelReaderWriter.Write<AvailabilitySetData>(Data, options, AzureSampleResourceManagerSampleContext.Default);
 
         AvailabilitySetData IPersistableModel<AvailabilitySetData>.Create(BinaryData data, ModelReaderWriterOptions options) => ModelReaderWriter.Read<AvailabilitySetData>(data, options, AzureSampleResourceManagerSampleContext.Default);
 
-        string IPersistableModel<AvailabilitySetData>.GetFormatFromOptions(ModelReaderWriterOptions options) => ((IPersistableModel<AvailabilitySetData>)Data).GetFormatFromOptions(options);
+        string IPersistableModel<AvailabilitySetData>.GetFormatFromOptions(ModelReaderWriterOptions options) => ((IPersistableModel<AvailabilitySetData>)DataDeserializationInstance).GetFormatFromOptions(options);
```

Comment:
Thought, for a Resource, the formal will always be JSON. It's not or this is for consistency?

Reply:
This matches the previous behavior and makes it consistent with the Data property which is being serialized.